api/src/main/java/org/pmiops/workbench/cohortbuilder/querybuilder/DemoQueryBuilder.java

```diff
@@ -0,0 +1,80 @@
+package org.pmiops.workbench.cohortbuilder.querybuilder;
+
+import com.google.cloud.bigquery.QueryParameterValue;
+import com.google.cloud.bigquery.QueryRequest;
+import org.pmiops.workbench.model.SearchParameter;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * DemoQueryBuilder is an object that builds {@link QueryRequest}
+ * for BigQuery for the following criteria types:
+ * DEMO_GEN, DEMO_AGE, DEMO_RACE and DEMO_DEC.
+ */
+@Service
+public class DemoQueryBuilder extends AbstractQueryBuilder {
+
+    private static final String DEMO_GEN =
+            "select distinct concat(cast(p.person_id as string), ',',\n" +
+                    "p.gender_source_value, ',',\n" +
+                    "p.race_source_value) as val\n" +
+                    "FROM `${projectId}.${dataSetId}.person` p\n" +
+                    "where p.gender_concept_id = @genderConceptId\n";
+
+    private static final String DEMO_AGE =
+            "select distinct concat(cast(p.person_id as string), ',',\n" +
+                    "p.gender_source_value, ',',\n" +
+                    "p.race_source_value) as val\n" +
+                    "FROM `${projectId}.${dataSetId}.person` p\n" +
+                    "where DATE_DIFF(CURRENT_DATE, DATE(p.year_of_birth, p.month_of_birth, p.day_of_birth), YEAR) = @age\n";
```

Comment:
In our metrics it's common to bucket participants into age ranges. Are we going to support that sort of thing? What actually shows up in the tree for age values?

Reply:
I think we will have a age or age range that you can select from the tree. So this will also have to handle a between ages as well. Just figure i would add that when it gets added in the ui stories.